src/server/pfs/cmds/cmds.go

```diff
@@ -71,13 +72,23 @@ or type (e.g. csv, binary, images, etc).`,
 				return err
 			}
 			defer c.Close()
-			_, err = c.PfsAPIClient.CreateRepo(
-				c.Ctx(),
-				&pfsclient.CreateRepoRequest{
-					Repo:        client.NewRepo(args[0]),
-					Description: description,
-				},
-			)
+
+			request := &pfsclient.CreateRepoRequest{
+				Repo:        client.NewRepo(args[0]),
+				Description: description,
+			}
+
+			txn, err := txncmds.GetActiveTransaction()
+			if err != nil {
+				return err
+			} else if txn != nil {
+				_, err = c.AppendCreateRepo(txn, request)
+				if err == nil {
+					fmt.Fprintf(os.Stderr, "Appended to transaction: %s\n", txn.ID)
+				}
+			} else {
+				_, err = c.PfsAPIClient.CreateRepo(c.Ctx(), request)
+			}
```

Comment:
Would it be possible to implement this server-side rather than clientside? We generally try to have as little logic as possible in the cmds, and it seems like this could be done pretty generically by checking the `ctx` to see if there's a txn in it or not server side.

Reply:
That sounds ok to me, worst that might happen is I end up with a circular dependency and need to shuffle some code around.  I'll give it a shot.